### PR TITLE
Modernise to run with python 2 or 3

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Run python-modernise over the sources, similar to https://github.com/ome/ome-model/pull/65.  Fix byte/string conversion manually, then fix any remaining flake8 issues (exception types primarily).

There may be additional undiscovered problems, but the only way to identify them is to release this, then test when converting downstream users like scc.  The primary thing to test here is that it doesn't cause any python 2 regressions.

Also note, as for ome-model, this drops Python 2.6 support which is a requirement for the `six` module used for Python 2.7-3.x compatibility.

### CODE REVIEW

**Comment by joshmoore** [comment]
Once the requested changes are approved, I assume this would be targeted as `0.2.0`.

File: yaclifw/version.py [hunk 1]
```diff
@@ -61,9 +63,9 @@ def call_git_describe(abbrev=4):
                    '--abbrev=%d' % abbrev], stdout=PIPE, stderr=PIPE)
         p.stderr.close()
         line = p.stdout.readlines()[0]
-        return line.strip()
+        return line.strip().decode("utf-8")
```
[the remark this answers is not in the document]
Comment (rleigh-codelibre) [reply yaclifw/version.py]: Once we drop Python 2 support, this can become `decode()`.

**Comment by rleigh-codelibre** [comment]
I've pushed a couple of changes to address @manics's suggestions.

**Comment by joshmoore** [comment]
Haven't heard any complaints. Releasing for downstream testing and we can increment the version as necessary.

**Pull request merged**